ui/components/app/confirm/info/row/text-token-units.tsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { BigNumber } from 'bignumber.js';
+
+import { calcTokenAmount } from '../../../../../../shared/lib/transactions-controller-utils';
+import {
+  formatAmount,
+  formatAmountMaxPrecision,
+} from '../../../../../pages/confirmations/components/simulation-details/formatAmount';
+import { ConfirmInfoRowText } from './text';
+
+type ConfirmInfoRowTextTokenUnitsProps = {
+  value: number | string | BigNumber;
+  decimals: number;
+};
+
+export const ConfirmInfoRowTextTokenUnits: React.FC<
+  ConfirmInfoRowTextTokenUnitsProps
+> = ({ value, decimals }) => {
+  const tokenValue = calcTokenAmount(value, decimals);
+
+  // FIXME - Precision may be lost for large values when using formatAmount
+  /** @see {@link https://github.com/MetaMask/metamask-extension/issues/25755} */
```

Comment:
This issue has been fixed, we can get rid of this comment now.

Reply:
yep! this is correct. It will be removed in another cherry-pick PR targeting the same release

Thanks for the comment 👍🏼

Reply:
Turns out we haven't cherry-picked the Precision fix to 12.1.0 yet. Cherry-pick PR https://github.com/MetaMask/metamask-extension/pull/26496